overlord/snapstate/snapmgr.go

```diff
@@ -484,6 +492,18 @@ func (m *SnapManager) ensureRefreshes() error {
 		return err
 	}
 
+	delta := timeutil.Next(refreshSchedule, lastRefresh)
+
+	m.nextRefresh = time.AfterFunc(delta, func() { m.doAutoRefresh(updated, tasksets) })
```

Comment:
maybe I am missing something but how can we reuse tasksets created based on current state we know now at a future point where the state can be completely different?